compiler/aarch64/codegen/OMRRegisterDependency.cpp

```diff
@@ -622,3 +622,94 @@ OMR::ARM64::RegisterDependencyConditions::stopUsingDepRegs(TR::CodeGenerator *cg
    if (_postConditions != NULL)
       _postConditions->stopUsingDepRegs(_addCursorForPost, ret1, ret2, cg);
    }
+
+void TR_ARM64ScratchRegisterDependencyConditions::addDependency(TR::CodeGenerator *cg, TR::Register *vr, TR::RealRegister::RegNum rr, uint8_t flag)
+   {
+   TR_ASSERT_FATAL(_numGPRDeps < TR::RealRegister::LastAssignableGPR - TR::RealRegister::FirstGPR + 1, "Too many GPR dependencies");
+
+   bool isGPR = rr <= TR::RealRegister::LastAssignableGPR;
+   TR_ASSERT_FATAL(isGPR, "Expecting GPR only");
+   if (!vr)
+      {
+      vr = cg->allocateRegister(TR_GPR);
+      cg->stopUsingRegister(vr);
+      }
+   _gprDeps[_numGPRDeps].setRegister(vr);
+   _gprDeps[_numGPRDeps].assignFlags(flag);
+   _gprDeps[_numGPRDeps].setRealRegister(rr);
+   ++_numGPRDeps;
+   }
+
+void TR_ARM64ScratchRegisterDependencyConditions::unionDependency(TR::CodeGenerator *cg, TR::Register *vr, TR::RealRegister::RegNum rr, uint8_t flag)
+   {
+   TR_ASSERT_FATAL(_numGPRDeps < TR::RealRegister::LastAssignableGPR - TR::RealRegister::FirstGPR + 1, "Too many GPR dependencies");
+
+   bool isGPR = rr <= TR::RealRegister::LastAssignableGPR;
+   TR_ASSERT_FATAL(isGPR, "Expecting GPR only");
+   TR_ASSERT_FATAL(vr, "Expecting non-null virtual register");
+
+   for (int i = 0; i < _numGPRDeps; i++)
+      {
+      if (_gprDeps[i].getRegister() == vr)
+         {
+         // Keep the stronger of the two constraints
+         //
+         TR::RealRegister::RegNum min = std::min(rr, _gprDeps[i].getRealRegister());
+         TR::RealRegister::RegNum max = std::max(rr, _gprDeps[i].getRealRegister());
+         if (min == TR::RealRegister::NoReg)
+            {
+            // Anything is stronger than NoReg
+            _gprDeps[_numGPRDeps].setRegister(vr);
```

Comment:
Is the index for `_gprDeps[]` here correct?

Reply:
This should be `i`. Thanks for pointing it out.